src/client/src/ui/status-connection/StatusButton.tsx

```diff
@@ -0,0 +1,80 @@
+import _ from 'lodash'
+import React, { Component, SyntheticEvent } from 'react'
+
+import { ConnectionState } from 'rt-system'
+import { ServiceConnectionStatus, ServiceStatus } from 'rt-types'
+import { Button, StatusCircle, StatusLabel, Root, AppUrl, ServiceListPopup, ServiceList } from './styled'
+import Service from './Service'
+
+interface State {
+  opened: boolean
+}
+
+export class StatusButton extends Component<
+  {
+    connectionStatus: ConnectionState
+    services: ServiceStatus[]
+  },
+  State
+> {
+  state = {
+    opened: false,
+  }
+
+  toggleOpen = (e: SyntheticEvent) => {
+    if (!this.isAppUrl(e.target)) {
+      this.setState(({ opened }) => ({ opened: !opened }))
+    }
+  }
+
+  isAppUrl = (element: any) => element instanceof HTMLInputElement
+
+  selectAll = (e: SyntheticEvent) => {
+    const input = e.target as HTMLInputElement
+    input.select()
```

Comment:
I'm not sure if the best practice is this way or with a ref. Let me know what you think.

Reply:
On the fence about this one. I did it this way because it's logical - and I didn't need to use the #ref syntax. What's your call on this?